src/Cli/dotnet/Commands/Solution/Add/SolutionAddCommand.cs

```diff
@@ -29,13 +28,21 @@ private static string GetSolutionFolderPathWithForwardSlashes(string path)
         return "/" + string.Join("/", PathUtility.GetPathWithDirectorySeparator(path).Split(Path.DirectorySeparatorChar, StringSplitOptions.RemoveEmptyEntries)) + "/";
     }
 
+    private static bool IsSolutionFolderPathInDirectoryScope(string relativePath)
+    {
+        return !string.IsNullOrEmpty(relativePath)
+            && !Path.IsPathRooted(relativePath) // This means path is in a different volume
+            && !relativePath.StartsWith(".."); // This means path is outside the solution directory
```

Comment:
It's probably not a real concern, but what happens if a symlink is used to go outside the scope?

Reply:
I am not sure if symlinks are supported on solution files, and AFAIK, neither our CLI nor VS Solution Persistence resolves symlinks before adding them to the solution.  (I could be wrong tho)

Reply:
You do a FixFilePath-type operation ahead of time, right?